src/main/java/me/contaria/seedqueue/SeedQueue.java

```diff
@@ -151,10 +151,31 @@ public static void discard(SeedQueueEntry entry) {
      */
     public static boolean shouldGenerate() {
         synchronized (LOCK) {
-            return getGeneratingCount() < getMaxGeneratingCount() && SEED_QUEUE.size() < config.maxCapacity;
+            return getGeneratingCount() < getMaxGeneratingCount() && hasMoreCapacity();
         }
     }
 
+    /**
+     * @return If the queue is not filled to capacity.
+     */
+    public static boolean hasMoreCapacity() {
+        return SEED_QUEUE.size() < config.maxCapacity;
+    }
+
+    /**
+     * @return If all {@link SeedQueueEntry} have reached the {@link SeedQueueConfig#maxWorldGenerationPercentage}.
+     */
+    public static boolean noPrioritizedRemaining() {
+        return SEED_QUEUE.stream().noneMatch(entry -> entry.isPrioritized() && !entry.isLocked());
+    }
+
+    /**
+     * @return If all {@link SeedQueueEntry} are not locked.
+     */
+    public static boolean noLockedRemaining() {
+        return SEED_QUEUE.stream().noneMatch(entry -> entry.isLocked() && !entry.isReady());
+    }
```

Comment:
Please use a for loop instead of streams, I've heard bad things about stream performance from people smarter than me and this is relatively hot code.
If you're feeling very motivated you could even remove the existing stream in `getGeneratingCount`

Reply:
That's rly interesting... Does java not decompose streams into simpler instructions like in actual functional languages? 🙃